### PR TITLE
[build][test] upgrade pyyaml for MXNET 1.8 version

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*Issue #, if available:*

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*

- Upgrade pyyaml to v5.4.1 due to CVE CVE-2020-14343 affected for versions below v5.4

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by saimidu — COMMENTED** [review]
Please also update the pyyaml version on TF 1.15 dockerfiles

File: test/dlc_tests/sanity/test_safety_check.py [hunk 1]
```diff
@@ -64,14 +62,12 @@
...
+            "py3": []
         },
         "training": {
             # for shipping pillow<=6.2.2 - the last available version for py2
             "py2": ['38449', '38450', '38451', '38452'],
-            # for shipping pyyaml v5.3.1 - blocked on upgrading to v5.4.1 due to dependency on awscli
-            "py3": ['39611']
+            "py3": []
         },
         "inference-neuron":{
```
Comment (saimidu) [inline test/dlc_tests/sanity/test_safety_check.py:72]: Please remove `'39611'` from the lists for inference-eia and inference-neuron as well.

**Comment by tejaschumbalkar** [comment]
> Please also update the pyyaml version on TF 1.15 dockerfiles

Addressed at https://github.com/aws/deep-learning-containers/pull/873

File: test/dlc_tests/sanity/test_safety_check.py (hunk 1 above)
Comment (tejaschumbalkar) [reply test/dlc_tests/sanity/test_safety_check.py]: Already removed in https://github.com/aws/deep-learning-containers/pull/860

File: mxnet/training/docker/1.8.0/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -11,6 +11,7 @@ ARG MX_URL=https://aws-mx-pypi.s3-us-west-2.amazonaws.com/1.8.0/aws_mx-1.8.0-py2
 # The smdebug pipeline relies for following format to perform string replace and trigger DLC pipeline for validating
 # the nightly builds. Therefore, while updating the smdebug version, please ensure that the format is not disturbed.
 ARG SMDEBUG_VERSION=0.9.4
+ARG OPENSSL_VERSION=1.1.1g
```
Comment (tejaschumbalkar) [inline mxnet/training/docker/1.8.0/py3/Dockerfile.cpu]: Keeping this as is. UL16 do not have `libssl1.1` package. ref: https://packages.ubuntu.com/search?suite=default&section=all&arch=i386&keywords=libssl&searchon=names

Comment (saimidu) [reply mxnet/training/docker/1.8.0/py3/Dockerfile.cpu]: In that case, could you change this to `OPENSSL_VERSION=1.1.1j`?

File: tensorflow/buildspec.yml [hunk 3]
```diff
@@ -1,82 +1,137 @@
-account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
-region: &REGION <set-$REGION-in-environment>
-framework: &FRAMEWORK tensorflow
-version: &VERSION 2.4.1
-short_version: &SHORT_VERSION 2.4
+  account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
+  region: &REGION <set-$REGION-in-environment>
+  framework: &FRAMEWORK tensorflow
+  version: &VERSION 2.3.1
+  short_version: &SHORT_VERSION 2.3
```
Comment (saimidu) [inline tensorflow/buildspec.yml:5]: Reviewer Note: To be reverted to get PR merge-ready.

File: mxnet/training/docker/1.8.0/py3/Dockerfile.cpu (hunk 2 above)
Comment (tejaschumbalkar) [reply mxnet/training/docker/1.8.0/py3/Dockerfile.cpu]: Added.

**Review by saimidu — COMMENTED** [review]
LGTM. Let's merge when all tests pass.

**Review by saimidu — COMMENTED** [review]
APPROVED.

Please revert config changes and enable test-skips.

**Review by saimidu — APPROVED** [review]

**Pull request merged**